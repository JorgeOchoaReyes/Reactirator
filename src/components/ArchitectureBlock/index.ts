export { TreeFolder } from './TreeFolder';
export { CreateComponent } from './CreateComponent';
export { CreateFolder } from './CreateFolder';