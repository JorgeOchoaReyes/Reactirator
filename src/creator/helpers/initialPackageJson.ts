export const initialPackageJsonCRA = {
  name: '',
  version: '0.1.0',
  description: '',
  private: true,
  dependencies: {
    '@testing-library/jest-dom': '',
    '@testing-library/react': '',
    '@testing-library/user-event': '',
    react: '',
    'react-dom': '',
    'react-scripts': '',
    'web-vitals': '',
  },
  scripts: {
    start: 'react-scripts start',
    build: 'react-scripts build',
    test: 'react-scripts test',
    eject: 'react-scripts eject',
  },
  eslintConfig: {
    extends: ['react-app', 'react-app/jest'],
  },
  browserslist: {
    production: ['>0.2%', 'not dead', 'not op_mini all'],
    development: ['last 1 chrome version', 'last 1 firefox version', 'last 1 safari version'],
  },
  devDependencies: {},
};

export const initialPackageJsonVite = {
  name: '',
  private: true,
  version: '0.1.0',
  type: 'module',
  scripts: {
    dev: 'vite',
    build: 'tsc && vite build',
    preview: 'vite preview',
  },
  dependencies: {
    react: '',
    'react-dom': '',
  },
  devDependencies: {
    '@types/react': '',
    '@types/react-dom': '',
    '@vitejs/plugin-react': '',
    typescript: '',
    vite: '',
  },
};
