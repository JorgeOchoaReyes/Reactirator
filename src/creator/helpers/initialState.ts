export const initialState = {
  appname: '',
  description: '',
  version: '0.1.0',
  typescript: false,
  prettier: false,
  flow: false,
  tailwind: false,
  bootstrap: false,
  reactbootstrap: false,
  materialui: false,
  styledcomponents: false,
  normalize: false,
  reactrouter: false,
  proptypes: false,
  sourcemapexplorer: false,
  storybook: false,
};

export const initialStateVite = {
  appname: '',
  description: '',
  version: '0.1.0',
  typescript: false,
  prettier: false,
  flow: false,
  tailwind: false,
  bootstrap: false,
  materialui: false,
  reactbootstrap: false,
  normalize: false,
  reactrouter: false,
  proptypes: false,
  sourcemapexplorer: false,
};