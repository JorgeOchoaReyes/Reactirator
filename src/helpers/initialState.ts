const initialState = {
    appname: '',
    typescript: false,
    prettier: false,
    tailwind: false,
    bootstrap: false,
    styledcomponents: false,
    normalize: false,
    reactrouter: false,
    propstype: false
};

export default initialState;