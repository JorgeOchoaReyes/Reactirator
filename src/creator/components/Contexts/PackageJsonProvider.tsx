import React, { ReactNode, useContext } from 'react';
import { actionJsonType } from '../../helpers/types';

type PackageContextType = {
  packageJson: any;
  dispatchJson: (object: actionJsonType) => void;
  baseSize: number;
} | null;

export const PackageContext = React.createContext<PackageContextType>(null);

export const PackageJsonProvider = ({
  children,
  packageJson,
  dispatchJson,
  baseSize,
}: {
  children: ReactNode;
  packageJson: any;
  dispatchJson: (object: actionJsonType) => void;
  baseSize: number;
}) => {
  return (
    <PackageContext.Provider value={{ packageJson, dispatchJson, baseSize }}>
      {children}
    </PackageContext.Provider>
  );
};

export const usePackageJson = () => {
  return useContext(PackageContext);
};
