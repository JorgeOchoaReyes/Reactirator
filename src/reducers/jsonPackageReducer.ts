import { actionJsonType } from '../helpers/types';

const jsonPackageReducer = (state: any, action: actionJsonType) => {
  switch (action.type) {
    case 'CHANGE_NAME':
      state.name = action.payload;
      return { ...state };
    case 'ADD':
      state[action.payload.category][action.payload.name] = action.payload.version;
      return { ...state };
    case 'REMOVE':
      delete state[action.payload.category][action.payload.name];
      return { ...state };
    case 'tailwind':
      return addTailwind(state, action.payload.version);
    case 'storybook':
      return addStorybook(state, action.payload.version);
    case 'prettier':
      return addPrettier(state, action.payload.version);
    case 'flow':
      return addFlow(state, action.payload.version);
    case 'bootstrap':
      return addBootstrap(state, action.payload.version);
    case 'sourcemapexplorer':
      return addSourcemapexplorer(state, action.payload.version);
    case 'normalize':
      return { ...state };
    case 'typescript':
      return addTypescript(state);
    default:
      throw new Error();
  }
};

export default jsonPackageReducer;

const addPrettier = (state: any, version: string) => {
  if (!state.devDependencies.prettier) {
    state.devDependencies['prettier'] = `${version}`;
  } else {
    delete state.devDependencies['prettier'];
  }
  return { ...state };
};

const addBootstrap = (state: any, version: string) => {
  if (!state.dependencies['bootstrap']) {
    state.dependencies['bootstrap'] = `^${version}`;
  } else {
    delete state.dependencies['bootstrap'];
  }
  return { ...state };
};

const addSourcemapexplorer = (state: any, version: string) => {
  if (!state.dependencies['source-map-explorer']) {
    state.dependencies['source-map-explorer'] = `^${version}`;
    state.scripts['analyze'] = "source-map-explorer 'build/static/js/*.js'";
  } else {
    delete state.dependencies['source-map-explorer'];
    delete state.scripts['analyze'];
  }
  return { ...state };
};

const addFlow = (state: any, version: string) => {
  if (!state.dependencies['flow-bin']) {
    state.dependencies['flow-bin'] = `^${version}`;
    state.scripts['flow'] = 'flow';
  } else {
    delete state.dependencies['flow-bin'];
    delete state.scripts['flow'];
  }
  return { ...state };
};

const addTypescript = (state: any): any => {
  if (!state.devDependencies['typescript']) {
    state.devDependencies['typescript'] = '^4.2.4';
    state.devDependencies['@types/jest'] = '^26.0.23';
    state.devDependencies['@types/node'] = '^12.20.11';
    state.devDependencies['@types/react'] = '^17.0.4';
    state.devDependencies['@types/react-dom'] = '^17.0.3';
  } else {
    delete state.devDependencies['typescript'];
    delete state.devDependencies['@types/jest'];
    delete state.devDependencies['@types/node'];
    delete state.devDependencies['@types/react'];
    delete state.devDependencies['@types/react-dom'];
  }
  return { ...state };
};

const addTailwind = (state: any, version: string): any => {
  if (!state.dependencies['@craco/craco']) {
    state.dependencies['@craco/craco'] = '^6.1.2';
    state.scripts.start = 'craco start';
    state.scripts.build = 'craco build';
    state.scripts.test = 'craco test';
    state.devDependencies['@tailwindcss/postcss7-compat'] = `^${version}`;
    state.devDependencies['autoprefixer'] = '^9.8.6';
    state.devDependencies['postcss'] = '^7.0.35';
    state.devDependencies['tailwindcss'] = `npm:@tailwindcss/postcss7-compat@^${version}`;
  } else {
    delete state.dependencies['@craco/craco'];
    state.scripts.start = 'npm start';
    state.scripts.build = 'npm build';
    state.scripts.test = 'npm test';
    delete state.devDependencies['@tailwindcss/postcss7-compat'];
    delete state.devDependencies['autoprefixer'];
    delete state.devDependencies['postcss'];
    delete state.devDependencies['tailwindcss'];
  }
  return { ...state };
};

const addStorybook = (state: any, version: string): any => {
  if (!state.scripts['storybook']) {
    state.scripts['storybook'] = 'start-storybook -p 6006 -s public';
    state.scripts['build-storybook'] = 'build-storybook -s public';
    state.devDependencies['@storybook/addon-actions'] = `^${version}`;
    state.devDependencies['@storybook/addon-essentials'] = `^${version}`;
    state.devDependencies['@storybook/addon-links'] = `^${version}`;
    state.devDependencies['@storybook/node-logger'] = `^${version}`;
    state.devDependencies['@storybook/preset-create-react-app'] = '^3.1.7';
    state.devDependencies['@storybook/addon-actions'] = `^${version}`;
    state.devDependencies['@storybook/react'] = `^${version}`;
  } else {
    delete state.scripts['storybook'];
    delete state.scripts['build-storybook'];
    delete state.devDependencies['@storybook/addon-actions'];
    delete state.devDependencies['@storybook/addon-essentials'];
    delete state.devDependencies['@storybook/addon-links'];
    delete state.devDependencies['@storybook/node-logger'];
    delete state.devDependencies['@storybook/preset-create-react-app'];
    delete state.devDependencies['@storybook/addon-actions'];
    delete state.devDependencies['@storybook/react'];
  }
  return { ...state };
};
