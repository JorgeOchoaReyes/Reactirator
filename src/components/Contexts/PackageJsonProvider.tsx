import React, { ReactNode, useContext, useEffect, useReducer, useState } from 'react';
import initialPackageJson from '../../helpers/initialPackageJson';
import jsonPackageReducer from '../../reducers/jsonPackageReducer';
import { actionJsonType } from '../../helpers/types';
import { getSizeOfPackagesList, searchPackages } from '../../services/package.service';

type PackageContextType = {
  packageJson: any;
  dispatchJson: (object: actionJsonType) => void;
  baseSize: number;
} | null;

export const PackageContext = React.createContext<PackageContextType>(null);

const PackageJsonProvider = ({ children }: { children: ReactNode }) => {
  const [packageJson, dispatchJson] = useReducer(jsonPackageReducer, initialPackageJson);
  const [baseSize, setBaseSize] = useState(0);

  const getVersionsOfBaseDeps = async (): Promise<any[]> => {
    const list = [];
    for (const ele in packageJson.dependencies) {
      const res = await searchPackages(ele, 1);
      list.push(res[0]);
      dispatchJson({
        type: 'ADD',
        payload: {
          category: 'dependencies',
          name: res[0].package.name,
          version: res[0].package.version,
        },
      });
    }
    return list;
  };

  const initializeTotalSize = async (listPkg: any[]): Promise<void> => {
    // initial size with only CRA dependencies
    const totalSize = await getSizeOfPackagesList(listPkg);
    const totalSizeInKb = Math.floor(totalSize / 1000);
    setBaseSize(totalSizeInKb);
  };

  useEffect(() => {
    (async () => {
      const updatedDepsList = await getVersionsOfBaseDeps();
      await initializeTotalSize(updatedDepsList);
    })();
  }, []);

  return (
    <PackageContext.Provider value={{ packageJson, dispatchJson, baseSize }}>
      {children}
    </PackageContext.Provider>
  );
};

export const usePackageJson = () => {
  return useContext(PackageContext);
};

export default PackageJsonProvider;
