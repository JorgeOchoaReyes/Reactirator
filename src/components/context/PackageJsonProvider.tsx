import React, { ReactNode, useContext, useEffect, useReducer, useState } from 'react';
import initialPackageJson from '../../helpers/initialPackageJson';
import jsonPackageReducer from '../../reducers/jsonPackageReducer';
import { actionJsonType } from '../../helpers/types';
import { getSizeOfPackagesList, searchPackages } from '../../services/package.service';

type PackageContextType = {
  packageJson: any;
  dispatchJson: (object: actionJsonType) => void;
} | null;

export const PackageContext = React.createContext<PackageContextType>(null);

const PackageJsonProvider = ({ children }: { children: ReactNode }) => {
  const [packageJson, dispatchJson] = useReducer(jsonPackageReducer, initialPackageJson);

  const getVersionsOfBaseDeps = async () => {
    Object.keys(initialPackageJson.dependencies).forEach((pkg) => {
      searchPackages(pkg, 1)
        .then((result) => {
          dispatchJson({
            type: 'ADD',
            payload: {
              category: 'dependencies',
              name: result[0].package.name,
              version: result[0].package.version,
            },
          });
        })
        .catch((error) => {
          console.log(error);
        });
    });
  };

  useEffect(() => {
    getVersionsOfBaseDeps();
  }, []);

  return (
    <PackageContext.Provider value={{ packageJson, dispatchJson }}>
      {children}
    </PackageContext.Provider>
  );
};

export const usePackageJson = () => {
  const { packageJson, dispatchJson } = useContext(PackageContext);
  const [baseSize, setBaseSize] = useState(0);

  const initializeTotalSize = async (): Promise<void> => {
    // initial size with only CRA dependencies
    const totalSize = await getSizeOfPackagesList(packageJson.dependencies);
    const totalSizeInKb = Math.floor(totalSize / 1000);
    setBaseSize(totalSizeInKb);
  };

  useEffect(() => {
    initializeTotalSize();
  }, []);

  return { packageJson, dispatchJson, baseSize };
};

export default PackageJsonProvider;
