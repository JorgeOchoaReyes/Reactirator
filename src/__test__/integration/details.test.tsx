/**
 * @jest-environment jsdom
 */
import React, { useReducer, useState } from 'react';
import { cleanup, render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import initialPackageJson from '../../creator/helpers/initialPackageJson';
import initialState from '../../creator/helpers/initialState';

import jsonPackageReducer from '../../creator/reducers/jsonPackageReducer';
import { PackageJsonProvider } from '../../creator/components/Contexts/PackageJsonProvider';
import { CardPackageJson } from '../../creator/components/PackageJsonBlock';
import { DetailsForm } from '../../creator/components/DetailsBlock';

afterEach(cleanup);

describe('change information about the project', () => {
  it('should render the details form', () => {
    const tree = render(<FakeDetailsPage />);
    expect(tree).toMatchSnapshot();
  });

  it('should change the app name', async () => {
    render(<FakeDetailsPage />);
    const inputEle = screen.getByDisplayValue('0.1.0');
    fireEvent.change(inputEle, { target: { value: '3.0.0' } });
    expect(screen.getByTestId('packagejson')).toHaveTextContent('"version": "3.0.0"');
  });

  it('should change the version', async () => {
    render(<FakeDetailsPage />);
    const inputEle = screen.getByPlaceholderText('Description');
    fireEvent.change(inputEle, { target: { value: 'blablablabla' } });
    expect(screen.getByTestId('packagejson')).toHaveTextContent('"description": "blablablabla"');
  });

  it('should change the description', async () => {
    render(<FakeDetailsPage />);
    const inputEle = screen.getByPlaceholderText('App. name');
    fireEvent.change(inputEle, { target: { value: 'My react app' } });
    expect(screen.getByTestId('packagejson')).toHaveTextContent('"name": "My react app"');
  });
});

const FakeDetailsPage = () => {
  const [input, setInput] = useState(initialState);
  const [packageJson, dispatchJson] = useReducer(
    jsonPackageReducer,
    JSON.parse(JSON.stringify(initialPackageJson))
  );
  return (
    <PackageJsonProvider packageJson={packageJson} dispatchJson={dispatchJson}>
      <DetailsForm input={input} setInput={setInput} />
      <CardPackageJson />
    </PackageJsonProvider>
  );
};
