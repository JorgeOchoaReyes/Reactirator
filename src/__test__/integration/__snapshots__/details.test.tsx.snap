// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add change information about the project should render the details page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-1/3 space-y-2"
      >
        <h3
          class="undefined block font-bold text-black text-center py-2"
        >
          Application information
        </h3>
        <div>
          <label
            class="block text-sm font-medium text-gray-700"
            for="appname"
          >
            Name
          </label>
          <div
            class="mt-1"
          >
            <input
              class=" w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
              id="appname"
              name="appname"
              placeholder="App. name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div>
          <label
            class="block text-sm font-medium text-gray-700"
            for="version"
          >
            Version
          </label>
          <div
            class="mt-1"
          >
            <input
              class="w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
              id="version"
              name="version"
              placeholder="Version"
              type="text"
              value="0.1.0"
            />
          </div>
        </div>
        <div>
          <label
            class="block text-sm font-medium text-gray-700"
            for="description"
          >
            Description
          </label>
          <div
            class="mt-1"
          >
            <textarea
              class="resize-none w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
              id="description"
              name="description"
              placeholder="Description"
            />
          </div>
        </div>
      </div>
      <div
        class="p-4 rounded text-xs bg-blueGray h-80 overflow-y-auto overflow-x-hidden shadow hover:shadow-lg transition duration-200 text-yellow-300"
        id="packagejson"
      >
        <pre
          data-testid="packagejson"
        >
          {
  "name": "",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "",
    "@testing-library/react": "",
    "@testing-library/user-event": "",
    "react": "",
    "react-dom": "",
    "react-scripts": "",
    "web-vitals": ""
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      "&gt;0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {}
}
        </pre>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-1/3 space-y-2"
    >
      <h3
        class="undefined block font-bold text-black text-center py-2"
      >
        Application information
      </h3>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="appname"
        >
          Name
        </label>
        <div
          class="mt-1"
        >
          <input
            class=" w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
            id="appname"
            name="appname"
            placeholder="App. name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="version"
        >
          Version
        </label>
        <div
          class="mt-1"
        >
          <input
            class="w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
            id="version"
            name="version"
            placeholder="Version"
            type="text"
            value="0.1.0"
          />
        </div>
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="description"
        >
          Description
        </label>
        <div
          class="mt-1"
        >
          <textarea
            class="resize-none w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
            id="description"
            name="description"
            placeholder="Description"
          />
        </div>
      </div>
    </div>
    <div
      class="p-4 rounded text-xs bg-blueGray h-80 overflow-y-auto overflow-x-hidden shadow hover:shadow-lg transition duration-200 text-yellow-300"
      id="packagejson"
    >
      <pre
        data-testid="packagejson"
      >
        {
  "name": "",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "",
    "@testing-library/react": "",
    "@testing-library/user-event": "",
    "react": "",
    "react-dom": "",
    "react-scripts": "",
    "web-vitals": ""
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      "&gt;0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {}
}
      </pre>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`change information about the project should render the details page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-1/3 space-y-2"
      >
        <h3
          class="undefined block font-bold text-black text-center py-2"
        >
          Application information
        </h3>
        <div>
          <label
            class="block text-sm font-medium text-gray-700"
            for="appname"
          >
            Name
          </label>
          <div
            class="mt-1"
          >
            <input
              class=" w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
              id="appname"
              name="appname"
              placeholder="App. name"
              type="text"
              value=""
            />
          </div>
        </div>
        <div>
          <label
            class="block text-sm font-medium text-gray-700"
            for="version"
          >
            Version
          </label>
          <div
            class="mt-1"
          >
            <input
              class="w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
              id="version"
              name="version"
              placeholder="Version"
              type="text"
              value="0.1.0"
            />
          </div>
        </div>
        <div>
          <label
            class="block text-sm font-medium text-gray-700"
            for="description"
          >
            Description
          </label>
          <div
            class="mt-1"
          >
            <textarea
              class="resize-none w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
              id="description"
              name="description"
              placeholder="Description"
            />
          </div>
        </div>
      </div>
      <div
        class="p-4 rounded text-xs bg-blueGray h-80 overflow-y-auto overflow-x-hidden shadow hover:shadow-lg transition duration-200 text-yellow-300"
        id="packagejson"
      >
        <pre
          data-testid="packagejson"
        >
          {
  "name": "",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "",
    "@testing-library/react": "",
    "@testing-library/user-event": "",
    "react": "",
    "react-dom": "",
    "react-scripts": "",
    "web-vitals": ""
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      "&gt;0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {}
}
        </pre>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-1/3 space-y-2"
    >
      <h3
        class="undefined block font-bold text-black text-center py-2"
      >
        Application information
      </h3>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="appname"
        >
          Name
        </label>
        <div
          class="mt-1"
        >
          <input
            class=" w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
            id="appname"
            name="appname"
            placeholder="App. name"
            type="text"
            value=""
          />
        </div>
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="version"
        >
          Version
        </label>
        <div
          class="mt-1"
        >
          <input
            class="w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
            id="version"
            name="version"
            placeholder="Version"
            type="text"
            value="0.1.0"
          />
        </div>
      </div>
      <div>
        <label
          class="block text-sm font-medium text-gray-700"
          for="description"
        >
          Description
        </label>
        <div
          class="mt-1"
        >
          <textarea
            class="resize-none w-full block text-center bg-white border border-gray-300 rounded-md py-2 px-3 focus:outline-none focus:ring-1 focus:ring-indigo-500 focus:border-indigo-500 text-sm"
            id="description"
            name="description"
            placeholder="Description"
          />
        </div>
      </div>
    </div>
    <div
      class="p-4 rounded text-xs bg-blueGray h-80 overflow-y-auto overflow-x-hidden shadow hover:shadow-lg transition duration-200 text-yellow-300"
      id="packagejson"
    >
      <pre
        data-testid="packagejson"
      >
        {
  "name": "",
  "version": "0.1.0",
  "description": "",
  "private": true,
  "dependencies": {
    "@testing-library/jest-dom": "",
    "@testing-library/react": "",
    "@testing-library/user-event": "",
    "react": "",
    "react-dom": "",
    "react-scripts": "",
    "web-vitals": ""
  },
  "scripts": {
    "start": "react-scripts start",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "browserslist": {
    "production": [
      "&gt;0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {}
}
      </pre>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
